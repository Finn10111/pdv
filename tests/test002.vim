edit test002.in

let doclines = [48, 46, 42, 38, 30, 27, 24, 20, 17, 13, 9, 6]

for line in doclines
	call cursor(line, 23)
	echo "Docline: " . line
	call pdv#DocumentLine()
endfor

call vimtest#SaveOut()
call vimtest#Quit()
