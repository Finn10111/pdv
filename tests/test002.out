<?php

abstract class TestClass
{

    /**
     * simplesFunc
     *
     */
    public function simplesFunc () {
    }

    /**
     * whiteSpaceNoParamsFunc
     *
     */
    public function whiteSpaceNoParamsFunc(  )
    {
    }

/**
 * simpleParamFunc
 *
 * @param mixed $parameter
 */
public function simpleParamFunc( $parameter )
{
}

    /**
     * paramFuncIntDefault
     *
     * @param int $parameter
     */
    public function paramFuncIntDefault( $parameter = 23 ) {
    }

  /**
   * multiParamsFuncFloatDefault
   *
   * @param int $someParam
   * @param mixed $anotherParam
   */
  public function multiParamsFuncFloatDefault( $someParam = 42.5, $anotherParam )
  {
  }

/**
 * typeHintParamsFunc
 *
 * @param mixed $foo
 * @param mixed $bar
 */
public function typeHintParamsFunc(array $foo, SomeClass $bar) {
}

    /**
     * complexDefaultFunc
     *
     * @param array $foo
     */
    public function complexDefaultFunc( array $foo = array( 1, 2, 3 ) ) {
    }

    /**
     * multiLineParamListFunc
     *
     * @param mixed $firstParam
     * @param mixed $secondParam
     * @param string $thirdParam
     * @param int $fourthParam
     */
    public function multiLineParamListFunc(
        FirstParam $firstParam = null,
        SecondParam $secondParam,
        array $thirdParam = array( "foo", array( "bar" ) ),
        $fourthParam = 42.23
    ) {
    }

    /**
     * protectedFunc
     *
     * @param mixed $foo
     * @param mixed $bar
     */
    protected function protectedFunc( $foo = null, array $bar )
    {
    }

    /**
     * privateStaticFunc
     *
     * @param mixed $foo
     */
    private static function privateStaticFunc( SomeClass $foo = null )
    {
    }

    /**
     * abstractPublicFunc
     *
     */
    abstract public function abstractPublicFunc();

    /**
     * protectedAbstractFunc
     *
     * @param string $foo
     */
    protected abstract function protectedAbstractFunc($foo="bar");

}
